source/exec/cache.d

```diff
@@ -31,23 +31,25 @@ class Cache: IExecProvider
 		import std.algorithm: map;
 		import std.array: array;
 		this.realExecProvider_ = realExecProvider;
-		this.allowedSources_ = sourceCodeWhitelist
-			.map!(x => x.getSourceCodeHash)
-			.array;
-		sort(this.allowedSources_);
-		assert(sourceCodeWhitelist.length == this.allowedSources_.length);
+		// allowedSources is no longer used
+		//this.allowedSources_ = sourceCodeWhitelist
+			//.map!(x => x.getSourceCodeHash)
+			//.array;
+		//sort(this.allowedSources_);
+		//assert(sourceCodeWhitelist.length == this.allowedSources_.length);
 	}
 
 	Tuple!(string, "output", bool, "success") compileAndExecute(RunInput input)
 	{
 		import std.range: assumeSorted;
 		import std.algorithm: canFind;
-		auto hash = getSourceCodeHash(input.source);
+		auto hash = getSourceCodeHash(input);
 
-		if (!assumeSorted(this.allowedSources_).canFind(hash)) {
-			auto result = realExecProvider_.compileAndExecute(input);
-			return result;
-		} else if (auto cache = hash in sourceHashToOutput_) {
+		// allowedSources is no longer used
+		//if (!assumeSorted(this.allowedSources_).canFind(hash)) {
+			//auto result = realExecProvider_.compileAndExecute(input);
+			//return result;
```

Comment:
This has been disabled for a long time already, I kept it as comments for now, but I doubt we will ever need it again.